src/features/edit/edit-label-ui.ts

```diff
@@ -193,24 +210,33 @@ export class EditLabelUI extends AbstractUIExtension {
         let height = 20;
 
         if (this.label) {
+            const zoom = getZoom(this.label);
             const bounds = getAbsoluteClientBounds(this.label, this.domHelper, this.viewerOptions);
-            x = bounds.x;
-            y = bounds.y;
-            height = bounds.height;
-            width = bounds.width + (this.additionalInputWidth * getZoom(this.label));
+            x = bounds.x + (this.label.editControlBoundsCorrection ? this.label.editControlBoundsCorrection.x * zoom : 0);
+            y = bounds.y + (this.label.editControlBoundsCorrection ? this.label.editControlBoundsCorrection.y * zoom : 0);
+            height = bounds.height + (this.label.editControlBoundsCorrection ? this.label.editControlBoundsCorrection.height * zoom : 0);
+            width = bounds.width + (this.label.editControlBoundsCorrection ? this.label.editControlBoundsCorrection.width * zoom : 0);
```

Comment:
These changes have some unintended side-effects for vertical editable labels (see edges in the class diagram example)

Reply:
Thanks, good catch! This is actually a tough one...
Previously, this just didn't cause problems "by accident" because we added a quite generous additional width and didn't apply the height to the input but only to the container (change below `this.editControl.style.height ...`).
We don't really know whether the label is rotated here. I guess the most pragmatic approach is to set a fixed height and width (as before this change) and let the label implementation decide if it wants to set height and width in a specific way (e.g. to the bounds). I'll make a suggestion in a new commit very soon.

Reply:
That makes sense